### PR TITLE
Start with memory: Pinecone

### PR DESCRIPTION
**Pull request by taksyon**
### Background #####################################################################

AutoGPT has not been able to start with vectors with Pinecone as a memory backend, but I may have solved this with a 4-file-change. 

### Changes ########################################################################
    
    modified:   autogpt/agent/agent.py
    modified:   autogpt/memory/pinecone.py
    modified:   autogpt/memory/base.py
    modified:   autogpt/memory/init.py

 base.py:
        Added save_vec_num and load_vec_num functions: line 10, 15
 pinecone.py:
        from-import load_vec_num, save_vec_num, clean_input: line 6, 7
        Added self.vec_num = load_vec_num(): line 20
        Added block to check index fullness percentage, prompt to clear: line 45
        Added save_vec_num(self.vec_num) after self.vec_num += 1: line 74
 agent.py:
        Added check for 100% index fullness before adding memory: line 242
 memory.init.py:
        Removed if init, memory.clear statement: line 50

### Documentation ####################################################################
        
        Changes are documented in the code with hash symbols

### Test Plan #########################################################################

        I tested my changes on both the master branch and stable-0.2.2 branch with successful results. However I will note that when AutoGPT creates an index for the first time it raises a service exception(503 error) and disconnects the pinecone API. But once you exit the program and restart, the 'auto-gpt' index is already created so it runs flawlessly. I plan on investigating this further tomorrow.
        

### PR Quality Checklist #################################################################

- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Comment by anbardoi** [comment]
The devcontainer.json was changed in the beginning to include my personal configuration settings for vs code. The second update for the devcontainer.json was to remove my configuration settings

**Comment by anbardoi** [comment]
removed the commit with my devcontainer.json configuration in the force-push

File: autogpt/agent/agent.py [hunk 1]
```diff
@@ -252,11 +253,27 @@ def start_interaction_loop(self):
...
+                    if index_fullness == 1:
+                        should_clear = clean_input(
+                            f"""Your Pinecone index is {Fore.RED}100% full{Style.RESET_ALL}.
+                                No new data can be stored.
+                                Clear index? (y/n):"""
+                        )
+                        if should_clear.lower() == "y":
+                            self.memory.clear
+                        else:
+                            pass
```
Comment (ntindle) [inline autogpt/agent/agent.py:301]: we should abstract this out a bit more to be compatible with any memory backend without a huge chain of if statements

Comment (ntindle) [reply autogpt/agent/agent.py:301]: This should probably live in the memory base class

File: autogpt/memory/base.py [hunk 2]
```diff
@@ -4,6 +4,20 @@
...
+
+def load_vec_num():
+    try:
+        with open("vec_num.txt", "r") as file:
+            vec_num = int(file.read())
+            return vec_num
+    except FileNotFoundError:
+        return 0
+
+
```
Comment (ntindle) [inline autogpt/memory/base.py]: These should be workspace dependent on their contexts cc: @collijk

**Comment by anbardoi** [comment]
Thanks for the insight, will work on this today

File: autogpt/memory/base.py (hunk 2 above)
Comment (taksyon) [reply autogpt/memory/base.py]: @ntindle I removed the save/load_vec_num functions in my latest commit. Now it accesses the vector count directly after initializing pinecone

File: autogpt/memory/pinecone.py [hunk 3]
```diff
@@ -41,26 +39,38 @@ def __init__(self, cfg):
...
-        # prompts user to clear or not based on preference.
+        # prompts user to clear or not based on preference; extracts vector count
+        # and assigns it to self.vec_num
+        # index fullness percentage and vector count are displayed in prompt
         index_stats = self.index.describe_index_stats()
+        namespaces = index_stats.get("namespaces", {})
+        default_namespace = namespaces.get("", {})
+        self.vec_num = default_namespace.get("vector_count", None)
+        if self.vec_num == None:
+            self.vec_num = "0"
```
Comment (anbardoi) [inline autogpt/memory/pinecone.py:50]: self.vec_num should be reset to an integer after the prompt. Fixing this today

**Comment by anbardoi** [comment]
Need to rework the check for a completely full index based on the new memory management system. I plan to abstract the method for other memory backends

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

**Pull request closed without merging**